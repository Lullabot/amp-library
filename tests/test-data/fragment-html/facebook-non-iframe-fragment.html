<div id="fb-root"></div>
<script>(function (d, s, id) {
    var js, fjs = d.getElementsByTagName(s)[0];
    if (d.getElementById(id)) return;
    js = d.createElement(s);
    js.id = id;
    js.src = "//connect.facebook.net/en_GB/sdk.js#xfbml=1&version=v2.6";
    fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));</script>

<div class="fb-post" data-href="https://www.facebook.com/zuck/posts/10101301165605491" data-width="500"
     data-show-text="true">
    <blockquote cite="https://www.facebook.com/zuck/posts/10101301165605491" class="fb-xfbml-parse-ignore"><p>As the
        world becomes more complex and governments everywhere struggle, trust in the internet is more important
        today...</p>Posted by <a href="https://www.facebook.com/zuck">Mark Zuckerberg</a> on&nbsp;<a
            href="https://www.facebook.com/zuck/posts/10101301165605491">Thursday, 13 March 2014</a></blockquote>
</div>

<div class="fb-video" data-href="https://www.facebook.com/facebook/videos/10153231379946729/" data-width="500"
     data-show-text="true">
    <blockquote cite="https://www.facebook.com/facebook/videos/10153231379946729/" class="fb-xfbml-parse-ignore"><a
            href="https://www.facebook.com/facebook/videos/10153231379946729/">How to Share With Just Friends</a>
        <p>How to share with just friends.</p>Posted by <a href="https://www.facebook.com/facebook/">Facebook</a> on
        Friday, 5 December 2014
    </blockquote>
</div>

<div class="fb-video" data-href="https://www.facebook.com/video.php?v=10153231379946729" data-width="500"
     data-show-text="true">
    <blockquote cite="https://www.facebook.com/video.php?v=10153231379946729" class="fb-xfbml-parse-ignore"><a
            href="https://www.facebook.com/video.php?v=10153231379946729">How to Share With Just Friends</a>
        <p>How to share with just friends.</p>Posted by <a href="https://www.facebook.com/facebook/">Facebook</a> on
        Friday, 5 December 2014
    </blockquote>
</div>

<div class="fb-video" data-href="https://www.facebook.com/zuck/videos/10103066366848051/" data-width="500" data-show-text="false">
    <blockquote cite="https://www.facebook.com/zuck/videos/10103066366848051/" class="fb-xfbml-parse-ignore">
    <a href="https://www.facebook.com/zuck/videos/10103066366848051/">Live from our Townhall Q&amp;A in Rome. Comment to ask a question!</a>
    <p>Live from our Townhall Q&amp;A in Rome. Comment to ask a question!</p>Posted by <a href="https://www.facebook.com/zuck">Mark Zuckerberg</a> on Monday, August 29, 2016
    </blockquote>
</div>

<div class="fb-post"
     data-href="https://www.facebook.com/photo.php?fbid=986393557241&amp;set=a.941146602501.2418915.4&amp;type=3"
     data-width="500" data-show-text="true">
    <blockquote cite="https://www.facebook.com/photo.php?fbid=986393557241&amp;set=a.941146602501.2418915.4&amp;type=3"
                class="fb-xfbml-parse-ignore">Posted by <a href="https://www.facebook.com/zuck">Mark Zuckerberg</a> on&nbsp;<a
            href="https://www.facebook.com/photo.php?fbid=986393557241&amp;set=a.941146602501.2418915.4&amp;type=3">Wednesday,
        September 14, 2011</a></blockquote>
</div>