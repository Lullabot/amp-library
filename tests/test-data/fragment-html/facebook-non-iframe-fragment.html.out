<div id="fb-root"></div>


<amp-facebook layout="responsive" data-href="https://www.facebook.com/zuck/posts/10101301165605491" data-embed-as="post" height="366" width="500"></amp-facebook>

<amp-facebook layout="responsive" data-href="https://www.facebook.com/facebook/videos/10153231379946729/" data-embed-as="post" height="366" width="500"></amp-facebook>

<amp-facebook layout="responsive" data-href="https://www.facebook.com/video.php?v=10153231379946729" data-embed-as="post" height="366" width="500"></amp-facebook>

<amp-facebook layout="responsive" data-href="https://www.facebook.com/zuck/videos/10103066366848051/" data-embed-as="video" height="366" width="500"></amp-facebook>

<amp-facebook layout="responsive" data-href="https://www.facebook.com/photo.php?fbid=986393557241&amp;set=a.941146602501.2418915.4&amp;type=3" data-embed-as="post" height="366" width="500"></amp-facebook>

ORIGINAL HTML
---------------
Line  1: <div id="fb-root"></div>
Line  2: <script>(function (d, s, id) {
Line  3:     var js, fjs = d.getElementsByTagName(s)[0];
Line  4:     if (d.getElementById(id)) return;
Line  5:     js = d.createElement(s);
Line  6:     js.id = id;
Line  7:     js.src = "//connect.facebook.net/en_GB/sdk.js#xfbml=1&version=v2.6";
Line  8:     fjs.parentNode.insertBefore(js, fjs);
Line  9: }(document, 'script', 'facebook-jssdk'));</script>
Line 10: 
Line 11: <div class="fb-post" data-href="https://www.facebook.com/zuck/posts/10101301165605491" data-width="500"
Line 12:      data-show-text="true">
Line 13:     <blockquote cite="https://www.facebook.com/zuck/posts/10101301165605491" class="fb-xfbml-parse-ignore"><p>As the
Line 14:         world becomes more complex and governments everywhere struggle, trust in the internet is more important
Line 15:         today...</p>Posted by <a href="https://www.facebook.com/zuck">Mark Zuckerberg</a> on&nbsp;<a
Line 16:             href="https://www.facebook.com/zuck/posts/10101301165605491">Thursday, 13 March 2014</a></blockquote>
Line 17: </div>
Line 18: 
Line 19: <div class="fb-video" data-href="https://www.facebook.com/facebook/videos/10153231379946729/" data-width="500"
Line 20:      data-show-text="true">
Line 21:     <blockquote cite="https://www.facebook.com/facebook/videos/10153231379946729/" class="fb-xfbml-parse-ignore"><a
Line 22:             href="https://www.facebook.com/facebook/videos/10153231379946729/">How to Share With Just Friends</a>
Line 23:         <p>How to share with just friends.</p>Posted by <a href="https://www.facebook.com/facebook/">Facebook</a> on
Line 24:         Friday, 5 December 2014
Line 25:     </blockquote>
Line 26: </div>
Line 27: 
Line 28: <div class="fb-video" data-href="https://www.facebook.com/video.php?v=10153231379946729" data-width="500"
Line 29:      data-show-text="true">
Line 30:     <blockquote cite="https://www.facebook.com/video.php?v=10153231379946729" class="fb-xfbml-parse-ignore"><a
Line 31:             href="https://www.facebook.com/video.php?v=10153231379946729">How to Share With Just Friends</a>
Line 32:         <p>How to share with just friends.</p>Posted by <a href="https://www.facebook.com/facebook/">Facebook</a> on
Line 33:         Friday, 5 December 2014
Line 34:     </blockquote>
Line 35: </div>
Line 36: 
Line 37: <div class="fb-video" data-href="https://www.facebook.com/zuck/videos/10103066366848051/" data-width="500" data-show-text="false">
Line 38:     <blockquote cite="https://www.facebook.com/zuck/videos/10103066366848051/" class="fb-xfbml-parse-ignore">
Line 39:     <a href="https://www.facebook.com/zuck/videos/10103066366848051/">Live from our Townhall Q&amp;A in Rome. Comment to ask a question!</a>
Line 40:     <p>Live from our Townhall Q&amp;A in Rome. Comment to ask a question!</p>Posted by <a href="https://www.facebook.com/zuck">Mark Zuckerberg</a> on Monday, August 29, 2016
Line 41:     </blockquote>
Line 42: </div>
Line 43: 
Line 44: <div class="fb-post"
Line 45:      data-href="https://www.facebook.com/photo.php?fbid=986393557241&amp;set=a.941146602501.2418915.4&amp;type=3"
Line 46:      data-width="500" data-show-text="true">
Line 47:     <blockquote cite="https://www.facebook.com/photo.php?fbid=986393557241&amp;set=a.941146602501.2418915.4&amp;type=3"
Line 48:                 class="fb-xfbml-parse-ignore">Posted by <a href="https://www.facebook.com/zuck">Mark Zuckerberg</a> on&nbsp;<a
Line 49:             href="https://www.facebook.com/photo.php?fbid=986393557241&amp;set=a.941146602501.2418915.4&amp;type=3">Wednesday,
Line 50:         September 14, 2011</a></blockquote>
Line 51: </div>


Transformations made from HTML tags to AMP custom tags
-------------------------------------------------------

<script> at line 2
 ACTION TAKEN: script facebook script tag was removed.

<div class="fb-post" data-href="https://www.facebook.com/zuck/posts/10101301165605491" data-width="500" data-show-text="true"> at line 12
 ACTION TAKEN: div.fb-post facebook javascript sdk embed code was converted to the amp-facebook tag.

<div class="fb-video" data-href="https://www.facebook.com/facebook/videos/10153231379946729/" data-width="500" data-show-text="true"> at line 20
 ACTION TAKEN: div.fb-video facebook javascript sdk embed code was converted to the amp-facebook tag.

<div class="fb-video" data-href="https://www.facebook.com/video.php?v=10153231379946729" data-width="500" data-show-text="true"> at line 29
 ACTION TAKEN: div.fb-video facebook javascript sdk embed code was converted to the amp-facebook tag.

<div class="fb-video" data-href="https://www.facebook.com/zuck/videos/10103066366848051/" data-width="500" data-show-text="false"> at line 37
 ACTION TAKEN: div.fb-video facebook javascript sdk embed code was converted to the amp-facebook tag.

<div class="fb-post" data-href="https://www.facebook.com/photo.php?fbid=986393557241&set=a.941146602501.2418915.4&type=3" data-width="500" data-show-text="true"> at line 46
 ACTION TAKEN: div.fb-post facebook javascript sdk embed code was converted to the amp-facebook tag.


AMP-HTML Validation Issues and Fixes
-------------------------------------
PASS

COMPONENT NAMES WITH JS PATH
------------------------------
'amp-facebook', include path 'https://cdn.ampproject.org/v0/amp-facebook-0.1.js'

